### ISSUE TITLE
test-gtx_fast_trigonometry fails on raspberry pi 3, while compiling with clang

### ISSUE THREAD

**Issue by kaomoneus** [g-truc/glm#839]
clang version 3.8.1-24+rpi (tags/RELEASE_381/final)
Affects glm-0.9.9.3

```
/home/pi/sync-dest/libglm/glm-0.9.9.3/test/gtx/gtx_component_wise.cpp:81:17: error: comparison of constant -128 with
      expression of type 'const char' is always false [-Werror,-Wtautological-constant-out-of-range-compare]
                        Error += A.y == -128 ? 0 : 1;
                                 ~~~ ^  ~~~~
1 error generated.
```

Looks like a clang bug (char variable **may** take a -128 value). And yet better to leave it here.

**Comment by kaomoneus** [comment]
Wow! Frankly I'm surprised about unsigned char.
In this case we should take into account CHAR_MIN and CHAR_MAX constants (defined in [limits](https://en.cppreference.com/w/cpp/types/climits)).

**Comment by Groovounet** [comment]
This issue should be fixed in master branch for GLM 0.9.9.4 release.

Thanks for contributing!
